client/src/partials/patients/edit/updatePatientGroups.modal.js

```diff
@@ -1,67 +1,68 @@
-// TODO Handle no patient groups case
 angular.module('bhima.controllers')
-.controller('UpdatePatientGroups', UpdatePatientGroups);
+  .controller('UpdatePatientGroups', UpdatePatientGroups);
 
-UpdatePatientGroups.$inject = ['$scope', '$uibModalInstance', 'PatientService', 'sessionPatient', 'sessionGroups', 'updateModel'];
+UpdatePatientGroups.$inject = [
+  '$uibModalInstance', 'PatientService', 'sessionPatient', 'sessionGroups',
+  'updateModel', 'NotifyService'
+];
 
-function UpdatePatientGroups($scope, $uibModalInstance, patients, sessionPatient, sessionGroups, updateModel) { 
+function UpdatePatientGroups($uibModalInstance, patients, sessionPatient, sessionGroups, updateModel, Notify) {
   var viewModel = this;
-  
+
   // TODO move to method
   viewModel.subscribedGroups = {};
-  sessionGroups.forEach(function (patientGroup) { 
-    viewModel.subscribedGroups[patientGroup.uuid] = true; 
+  sessionGroups.forEach(function (patientGroup) {
+    viewModel.subscribedGroups[patientGroup.uuid] = true;
   });
 
   // TODO Handle errors with generic modal exception display (inform system administrator)
   patients.groups()
-    .then(function (result) { 
-      viewModel.patientGroups = result;
-    });
+    .then(function (groups) {
+      viewModel.patientGroups = groups;
+    })
+    .catch(Notify.handleError);
+
+  viewModel.confirmGroups = function confirmGroup(groupForm) {
 
-  viewModel.confirmGroups = function confirmGroup() { 
-    var formIsUpdated = $scope.groupForm.$dirty;
-    
-    // Simply exit the modal
-    if (!formIsUpdated) { 
-      closeModal(); 
+    // if untouched, exit the modal
+    if (groupForm.$pristine) {
+      closeModal();
       return;
     }
 
-    patients.updateGroups(sessionPatient.uuid, viewModel.subscribedGroups)
-      .then(function () { 
+    return patients.updateGroups(sessionPatient.uuid, viewModel.subscribedGroups)
+      .then(function () {
 
         // TODO move to method
         var formatControllerResponse = [];
 
-        // Fetch each of the updated group definitions and collect them in an array 
-        Object.keys(viewModel.subscribedGroups).forEach(function (groupKey) { 
-          
-          if (viewModel.subscribedGroups[groupKey]) { 
+        // Fetch each of the updated group definitions and collect them in an array
+        Object.keys(viewModel.subscribedGroups).forEach(function (groupKey) {
+          if (viewModel.subscribedGroups[groupKey]) {
             formatControllerResponse.push(fetchGroupObject(groupKey));
           }
         });
-        
+
         updateModel(formatControllerResponse);
         closeModal();
       });
   };
 
   // TODO Refactor - use stores?
-  function fetchGroupObject(uuid) { 
+  function fetchGroupObject(uuid) {
     var groupObject;
-    viewModel.patientGroups.some(function (patientGroup) { 
+    viewModel.patientGroups.some(function (patientGroup) {
```

Comment:
The `while(i--)` syntax cannot be applied here ? i think it is more clear, what do you think about that ?